datafusion/proto/tests/cases/roundtrip_logical_plan.rs

```diff
@@ -1592,9 +1591,12 @@ fn roundtrip_aggregate_udf() {
 
 #[test]
 fn roundtrip_scalar_udf() {
-    let fn_impl = |args: &[ArrayRef]| Ok(Arc::new(args[0].clone()) as ArrayRef);
-
-    let scalar_fn = make_scalar_function(fn_impl);
+    let scalar_fn = Arc::new(|args: &[ColumnarValue]| {
+        let ColumnarValue::Array(array) = &args[0] else {
+            panic!()
+        };
+        Ok(ColumnarValue::Array(Arc::new(array.clone()) as ArrayRef))
```

Comment:
I wonder if we could make this code easier to work with using `From` impls, like

```suggestion
        Ok(ColumnarValue::from(Arc::new(array.clone()) as ArrayRef))
```

Maybe it doesn't matter 🤔

Reply:
Okay, I added two `From` impls for `ColumnarValue`.